packages/rum-core/src/domain/action/getActionNameFromElement.spec.ts

```diff
@@ -428,6 +478,15 @@ describe('getActionNameFromElement', () => {
       expect(nameSource).toBe('text_content')
     })
 
+    it('removes only the child with programmatic action name in textual content', () => {
+      const { name, nameSource } = getActionNameFromElement(
+        appendElement('<div>Foobar <div data-dd-action-name="custom action">bar<div></div>'),
+        defaultConfiguration
+      )
+      expect(name).toBe('Foobar')
```

Comment:
The difference between the buggy implementation and the working one comes down to a single space here, which might be a bit confusing when debugging this in the future. I think it'd be worth adding a bit more text to make the difference more clear:
```suggestion
      const { name, nameSource } = getActionNameFromElement(
        appendElement('<div>Foobar Baz<div data-dd-action-name="custom action">bar<div></div>'),
        defaultConfiguration
      )
      expect(name).toBe('Foobar Baz')
```
With this version, the buggy implementation would yield `Foo Bazbar`, which makes it clearer what the problem is.